onnxruntime/core/providers/cpu/signal/dft.cc

```diff
@@ -185,11 +185,10 @@ template <typename T>
 T next_power_of_2(T in) {
   in--;
   T out = 1;
-  while (out < in) {
-    in |= in >> out;
+  while (out <= in) {
```

Comment:
>while (out <= in) { [](http://example.com/codeflow?start=2&length=19)

Question: So if the input is 2, then the expected return is 4, right?

Reply:
2 should remain 2, since it is already a power of 2.

Reply:
Ah, different interpretation of the word "next" :b. I was thinking that if Bob is sitting on the train at the Seattle train stop, then the next train stop is Tukwila, not the stop where Bob is already at :b.